src/components/Details/Details.js

```diff
@@ -3,29 +3,39 @@ import About from './Tabs/About';
 import BaseStats from './Tabs/BaseStats';
 import Evolution from './Tabs/Evolution';
 import './Details.css';
+import { usePokemons } from '../../context/PokemonsProvider';
+import { usePokemonImage } from '../../hooks';
 
-function Details( { pokemon, setPokemonData } ) {
-	const [ tab, setTab ] = useState( 'about' );
+const TAB_ABOUT = 'about';
+const TAB_STATS = 'base-stats';
+const TAB_EVOLUTION = 'evolution';
+const TAB_DEFAULT = TAB_ABOUT;
 
-	// get pokemon image
-	const sprites = pokemon.sprites.other;
-	const imgURL = sprites.dream_world.front_default || sprites[ 'official-artwork' ].front_default;
+function Details( { pokemon } ) {
+	const [ currentTab, setCurrentTab ] = useState( TAB_DEFAULT );
+	const { setCurrentPokemonId } = usePokemons();
```

Comment:
By using this "custom hook" (which is basically a useConetext) there is a re-render of this entire component each time that pokemon changes. I think that it's better to use the "usePokemons" only inside the <Evolution> component (and the Evolution component will take care of the tab change as part of a prop that will be passed to it).

Reply:
Done